pkg/scheduler/core/generic_scheduler.go

```diff
@@ -207,13 +207,26 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 		return result, err
 	}
 
+	// TODO - Convert all built-in predicates to Plugin in the future.
+	// Pass the result of RunFilterPlugins to all built-in predicates for
+	// getting a smaller feasible node set
+	pluginFilteredNodes, pluginFilterFailedMap := g.framework.RunFilterPlugins(pluginContext, pod, nodes)
```

Comment:
As you noted yourself, calling filter plugins before `findNodesThatFit` does not work either, because `findNodesThatFit`  goes over all nodes. You should let `findNodesThatFit` call `RunFilterPlugins`. The third argument of `RunFilterPlugins` should be a single node.

Reply:
@bsalamat 

I worked on it yesterday.

I still called RunFilterPlugins before calling findNodesThatFit.
Then, a node name map is generated basing on the result of RunFilterPlugins.
If the node can be found in above map, all predicates are called to evaluated it.

I think the framework would call each extension point's plugin in the future.
If we follow predicate's approach to invoke Filter on each node, it will break the integrity of framework.

BTW, the new diff is attached and please help review it.

Reply:
Why do you think the scheduler calling `RunFilterPlugins` for each node will break the integrity of the framework? Right now the scheduler has a mechanism of parallelizing the calls for each predicate. If we had called `RunFilterPlugins` inside `fineNodesThatFit` and for each node, we could have used the existing parallelism, but ithis PR you had to replicate that logic inside `RunFilterPlugins`.
Moreoever, `findNodesThatFit` looks a bit odd now. We build a map of nodes that have passed `RunFilterPlugins` and still go over all nodes, start a go routine for each one of those nodes and return immediately if the node has not passed `RunFilterPlugins`. I think none of these were necessary if `findNodesThatFit` called `RunFilterPlugins` for each node.

Reply:
I think the framework calls each extension's handler one by one in the near future.
        Pre-filter ,  Filter, Post-filter, Scoring, 
        Normalize scoring, Reserve, Permit, 
       Pre-bind, Bind, Post-bind
       Un-reserve
If we use this approach, the framework is clean and the readability is good.

I guess **fineNodesThatFit** will be replaced, and this is the reason that I duplicate logic of this API.

Anyway, I will follow your proposal in this PR.

Reply:
@goodluckbot 
   Framework doesn't handle any feature.
   On the contrary, it should be handled by each individual Plugin.

   In the future, each built-in predicate should be converted into Plugin.

Reply:
You are right.
In my previous implementation,  RunFilterPlugins iterates all nodes to figure out all feasible nodes.